docs/src/man/backends.md

```diff
@@ -0,0 +1,131 @@
+
+## Allocators
+
+Evaluating complex tensor networks is typically done most efficiently by pairwise operations.
+As a result, this procedure often requires the allocation of many temporary arrays, which can affect performance for certain operations.
+To mitigate this, `TensorOperations` exposes an allocator system, which allows users to more finely control the allocation of both output tensors and temporary tensors.
+
+In particular, the allocator system is used in multiple ways:
+As mentioned before, it can be used to allocate and free the intermediate tensors that are required to evaluate a tensor network in a pairwise fashion.
+Additionally, it can also be used to allocate and free temporary objects that arise when reshaping and permuting input tensors, for example when making them compatible with BLAS instructions.
+
+### Allocator Selection
+
+The allocator system can only be accessed *locally*, by passing an allocator to the `@tensor` macro, the `ncon` function, or the function-based interface.
+
+```julia
+using TensorOperations
+myallocator = ManualAllocator()
+
+# inserting a backend into the @tensor macro
+@tensor allocator = myallocator A[i,j] := B[i,k] * C[k,j]
+
+# inserting an allocator into the ncon function
+D = ncon([A, B, C], [[1, 2], [2, 3], [3, 1]]; allocator=myallocator)
+
+# inserting a backend into the function-based interface
+tensoradd(A, pA, conjA, B, pB, conjB, α, β, DefaultBackend(), myallocator)
+```
+
+Important to note here is that the backend system is prioritized over the allocator system.
+In particular, this means that the backend will be selected **first**, while only then the allocator should be inserted.
+
+### Available Allocators
+
+`TensorOperations` also provides some options for allocators out-of-the box.
+
+```@docs
+DefaultAllocator
+ManualAllocator
+```
+
+By default, the `DefaultAllocator` is used, which uses Julia's built-in memory management system.
+Optionally, it can be useful to use the `ManualAllocator`, as the manual memory management reduces the pressure on the garbage collector.
+In particular in multi-threaded applications, this can sometimes lead to a significant performance improvement.
+
+Finally, users can also opt to use the `Bumper.jl` system, which pre-allocates a slab of memory that can be re-used afterwards.
+This is available through a package extension for `Bumper`.
+Here, the `allocator` object is just the provided buffers, which are then used to store the intermediate tensors.
+
+```julia
+using TensorOperations, Bumper
+buf = Bumper.default_buffer()
+@no_escape buf
+    @tensor allocator = buf A[i,j] := B[i,k] * C[k,j]
+end
+```
+For convenience, the construction above is also provided in a specialized macro form which is fully equivalent:
+
+```@docs
+@butensor
```

Comment:
We might need to check if this works, I don't think this macro is currently exported. (We could also export it)